src/cargo/core/package.rs

```diff
@@ -106,6 +106,8 @@ pub struct SerializedPackage {
     metabuild: Option<Vec<String>>,
     default_run: Option<String>,
     rust_version: Option<RustVersion>,
+    include: Vec<String>,
```

Comment:
While it might be useful to have `include` and `exclude` in `cargo metadata` output, it doesn't seem to be a proper way to fix rust-lang/rust-clippy#11677. The solution might end up reimplement the logic [Cargo find and package files](https://github.com/rust-lang/cargo/blob/7bb7b539558dc88bea44cee4168b6269bf8177b0/src/cargo/sources/path.rs#L133-L200), which includes `.gitignore`, shell glob syntax, symlinks handling, git dirty status, and other details (see #11405). `cargo package --list` might be more reliable, but not 100% accurate (see #8407 and #11666).

I would suggest starting the discussion from an issue, and jump back to implementation when the design is settled.

Reply:
Yes that's a good point. I'll open an issue so it can be discussed.

Reply:
I opened https://github.com/rust-lang/cargo/issues/13331.